src/docs/cookbook-events/like-feature/index.md

```diff
@@ -0,0 +1,527 @@
+  if (currentLikes === null) return previousLikes;
+  else return previousLikes.length < currentLikes.length ? getDifference(currentLikes, previousLikes) : getDifference(previousLikes, currentLikes);
+}
+const getLikeCount = (previousLikes, currentLikes) => {
+  return previousLikes === null || previousLikes.length < currentLikes.length ? 1 : -1;
+}
+// Reducer returns an object with listing ID's as keys and amount of likes as values
+const sortEvents = events => {
+  return likesToBeUpdated = events.reduce((likes, event) => {
+    const { resource: user, previousValues } = event.attributes;
+    const { likedListings: previouslyLikedListings} = previousValues.attributes.profile.privateData || {};
+    const likedListings = user.attributes.profile?.privateData?.likedListings; 
+    const likeCount = getLikeCount(previouslyLikedListings, likedListings); 
+    const listingId = getLikedListingId(previouslyLikedListings, likedListings); 
+    likes[listingId] = likes[listingId] ? likes[listingId] + likeCount : likeCount;
+    return likes;
+  }, {})
+}
+```
+Finally, to call the right functions, let's make a few changes to the `pollLoop` function:
+
+```diff
+const pollLoop = (sequenceId) => {
+  var params = sequenceId ? {startAfterSequenceId: sequenceId} : {createdAtStart: startTime};
+  queryEvents(params)
+    .then(res => {
+      const events = res.data.data;
+      const lastEvent = events[events.length - 1];
+      const fullPage = events.length === res.data.meta.perPage;
+      const delay = fullPage? pollWait : pollIdleWait;
+-    const lastSequenceId = lastEvent ? lastEvent.attributes.sequenceId : sequenceId;
+-     	      events.forEach(e => {	
+-       analyzeEvent(e);	
+-     });
+-      if (lastEvent) saveLastEventSequenceId(lastEvent.attributes.sequenceId);	
+-     setTimeout(() => {pollLoop(lastSequenceId);}, delay);
++     const likesToUpdate = sortEvents(events);
++     const actions = Object.keys(likesToUpdate).map(key => updateListing(key, likesToUpdate[key]));
++     const results = Promise.all(actions);
++     results.then(result => {
++       result.forEach(el => {
++         console.log(`Listing ID ${el.data.data.id.uuid} now has ${el.data.data.attributes.publicData.likes} like(s).`)
++       })
++
++       if (lastEvent) saveLastEventSequenceId(lastEvent.attributes.sequenceId);
++       setTimeout(() => {pollLoop(lastSequenceId);}, delay);
++     })
++   });
+};
+```
+
+You can now run the script locally following the instructions [here](https://github.com/sharetribe/flex-integration-api-examples#getting-started).
+
+## Increment the counter in the UI
+You can now click on the likes button and the amount of likes will be updated to listing extended data. However, the like counter doesn't get updated immediately due to the slight latency in our events listener. We can add a increment the counter using React state. In addition, we'll add conditional styling to reflect has the listing been liked or not.
```

Comment:
I think this could elaborate a bit more on the idea that the likes will be eventually consistent and that for UI reasons will update the like count also in the front end so that the user will know that the listing is liked but the actual change happens via events. I know it's already mentioned there but I think it could be explained in a bit more detail so that readers who are not so familiar with this kind of "eventually consistent" data flow will understand the idea of this artificial state change in the UI.

Reply:
And discuss that this is not totally necessary but what would be the downsides of not doing this UI state change. And on the other hand, what problems may this introduce if the poll delay in the events listener is long and it takes a long time for the likes data to update in the API.